app/src/main/java/app/revanced/manager/ui/viewmodel/MainViewModel.kt

```diff
@@ -19,12 +35,40 @@ class MainViewModel(
     private val patchBundleRepository: PatchBundleRepository,
     private val patchSelectionRepository: PatchSelectionRepository,
     private val keystoreManager: KeystoreManager,
+    private val reVancedAPI: ReVancedAPI,
+    private val app: Application,
     val prefs: PreferencesManager
 ) : ViewModel() {
+    var updatedManagerVersion: String? by mutableStateOf(null)
+        private set
+
+    init {
+        viewModelScope.launch { checkForUpdates() }
+    }
+
+    fun dismissUpdateDialog() {
+        updatedManagerVersion = null
+    }
+
+    private suspend fun checkForUpdates() {
+        if (!prefs.managerAutoUpdates.get()) return
```

Comment:
Check if the network is down or metered like this:
https://github.com/ReVanced/revanced-manager/blob/1a83315424f3b2d50d36e2e963992b20ba9e4a5d/app/src/main/java/app/revanced/manager/domain/repository/PatchBundleRepository.kt#L135